config/supervisord.conf

```diff
@@ -17,7 +17,7 @@ stderr_logfile=/dev/fd/2
 stderr_logfile_maxbytes=0
 
 [program:ramcloud-server]
-command=/usr/local/bin/rc-server --externalStorage %(ENV_RC_EXTERNAL_STORAGE)s --clusterName %(ENV_RC_CLUSTER_NAME)s --local basic+udp:host=%(ENV_RC_IP)s,port=11112 --replicas 1 --usePlusOneBackup true
+command=/usr/local/bin/rc-server --externalStorage %(ENV_RC_EXTERNAL_STORAGE)s --clusterName %(ENV_RC_CLUSTER_NAME)s --local basic+udp:host=%(ENV_RC_IP)s,port=11112 --replicas 1 --usePlusOneBackup true --backupInMemory --hugepage
```

Comment:
Not that I am suggesting it for production, but it would be interesting to know what effect --usePlusOneBackup false has on performance.

Reply:
We don't want to use usePlusOneBackup on prod. 

So when a table creation is performed on RAMCloud, a zookeeper entry with the corresponding table id & chosen server for master copy is saved in zookeeper. I.e., we know which server got chosen to store the master copy, but no mention of servers with backup copies are mentioned in zookeeper. The "default" behavior of RAMCloud on prod is to select the server(s) for backup copies randomly among those not chosen to store the master copy. This is acceptable behavior on prod.

However, for testing, this means we need to depend on the "random seed" to know where the backup copy is, or use the usePlusOneBackup argument. This argument means that, if server id 2 is chosen for the master copy, we always know that server id 3 will be chosen for the backup copy (server id for backup is always the server id for master "plus one", and yes there is wraparound for "plus one"... I'm the person that introduced this argument to RAMCloud, it was the last checkin of the RAMCloud repo before it got retired forever...). This gives us a predictable way in tests to intentionally take down a server containing the backup copy, useful for tests...